expandFns.php

```diff
@@ -221,6 +221,17 @@ function title_capitalization($in, $caps_after_punctuation) {
     create_function('$matches', 'return mb_strtolower($matches[0]);'),
     trim($new_case)
   );
+  
+  $new_case = preg_replace_callback("/ (L')([a-zÀ-ÿ]+)/", function($matches) { /** French l'Words  **/
+    return " l'" . mb_ucfirst($matches[2]);
+  }, ' ' . $new_case);
+  $new_case = mb_ucfirst(trim($new_case));
+
+  $new_case = preg_replace_callback("/ (D')([a-zÀ-ÿ]+)/", function($matches) { /** French d'Words  **/
```

Comment:
You can use \b to match a word boundary, i.e. `\b(D').  And there's no need for brackets around the (D') / (L') as if there is a match, we don't need to refer to $matches to figure out what it will be!